ACPs/77-reinventing-subnets/README.md

```diff
@@ -127,18 +149,27 @@ type EvictSubnetValidatorTx struct {
 The `Message` field in the above transaction must be an Avalanche Warp Message using the `AddressedCall` format with the below payload. A `typeID` for distinguishing between the different payload types on the P-Chain must be included prior to this ACP being considered "Implementable".
 
 ```text
-+-----------+----------+----------+
-|      txID : [32]byte | 32 bytes |
-+-----------+----------+----------+
-                       | 32 bytes |
-                       +----------+
++----------+----------+----------+
+| subnetID : [32]byte | 32 bytes |
++----------+----------+----------+
+|   nodeID : [32]byte | 32 bytes |
++----------+----------+----------+
+|   weight :   uint64 |  8 bytes |
++----------+----------+----------+
+|    nonce :   uint64 |  8 bytes |
```

Comment:
seems odd to me to use `nonce` for one message and `expiry` for another?

Reply:
is this nonce unique over `subnetID`/`nodeID`? seems to only be over `nodeID` right now?

Reply:
nonce is unique over `subnetID`/`nodeID`. Clarified.

Using `nonce` in `RegisterSubnetValidatorTx` seems strange since it would require the validator additions to be posted in-order onto the P-Chain. If we want to support issuing nonce of `2` after issuing nonce of `3`, we'd need to track all used nonces which I don't think is ideal.

For validator updates, we only have to keep track of the most recently used `nonce` for each `subnetID`/`nodeID`

Reply:
> Using nonce in RegisterSubnetValidatorTx seems strange since it would require the validator additions to be posted in-order onto the P-Chain.

I feel like this assumption exists regardless? Register must be posted before updates to its weight can be made?

Reply:
We do want ordering for a single validator (`subnetID`/`nodeID`). We want it to be register -> set weight nonce `0` -> set weight nonce `5` -> etc.

We do not want ordering for multiple validators (register validator `1` -> register validator `2` -> register validator `3`). Validator `3` should be able to register before validator `1` and `2` without preventing `1` and `2` from registering.

Doing a `nonce` for set weight guarantees the ordering for a single validator. Doing `expiry` guarantees no ordering for multiple validators.

Reply:
> We do want ordering for a single validator (subnetID/nodeID). We want it to be register -> set weight nonce 0 -> set weight nonce 5 -> etc.

If the same validator restakes, does that mean it goes back to 0? (seems like that could replay)

Reply:
I've clarified this part in the specification in the latest commit [196417a](https://github.com/avalanche-foundation/ACPs/pull/92/commits/196417aea8b1403fbc7c1d1d5f09fb320f080a08)